src/main/scala/edu/ie3/simona/agent/grid/ReceivedValuesStore.scala

```diff
@@ -100,27 +100,29 @@ object ReceivedValuesStore {
     /* Collect everything, that I expect from my asset agents */
     val assetsToReceivedPower: NodeToReceivedPower = nodeToAssetAgents.collect {
       case (uuid: UUID, actorRefs: Set[ActorRef]) =>
-        (uuid, actorRefs.toVector.map(actorRef => actorRef -> None))
+        (uuid, actorRefs.map(actorRef => actorRef -> None).toMap)
     }
 
-    /* Add everything, that I expect from my sub ordinate grid agents. Build distinct pairs of sending actor reference
-     * and target node */
-    inferiorSubGridGateToActorRef.toVector
+    /* Add everything, that I expect from my sub ordinate grid agents.
+     * Build distinct pairs of sending actor reference and target node.
+     * Convert to sequence first, since the map operation conflates
+     * key/value pairs with the same key */
```

Comment:
Great question. 

There can be multiple connections between two subgrids, which up until #25 requires multiple messages to be sent - which wasn't really handled though.
The thing here is that the DBFS tests that we currently have are not able to perform here, since TestProbes are not used and just once test actor acts as a stand-in for multiple GridAgents of inferior grids. I think this is why we need `toSeq` here.

The whole situation is really not satifying, but for this issue here I'd suggest waiting for #25 to enable multiple connections between subgrids. I want to improve the tests there to include said multiple connections and then check again if this code can be improved still (I think so). 
Implementing this in this PR would be double the work, since with #25 the behavior changes again.

Reply:
Pls check how I refactored this in #25: https://github.com/ie3-institute/simona/blob/2ad69c43703ebae16bd47987476a3dc668c1ea31/src/main/scala/edu/ie3/simona/agent/grid/ReceivedValuesStore.scala#L107-L127

Reply:
Also, the reason why this works is because yesterday in #25, I adapted the test to use multiple inferior-GA-TestProbes

https://github.com/ie3-institute/simona/blob/2ad69c43703ebae16bd47987476a3dc668c1ea31/src/test/scala/edu/ie3/simona/agent/grid/DBFSAlgorithmCenGridSpec.scala

Reply:
Alright thanks for the feedback I agree lets then handle this in #25